### PR TITLE
feat(oci-layout): support in fetch and push for manifest and blob commands

### PR DESCRIPTION
**Pull request by qweeah**
Related to #378

### CODE REVIEW

File: cmd/oras/blob/push.go [hunk 1]
```diff
@@ -48,30 +47,33 @@ func pushCmd() *cobra.Command {
 
 ** This command is in preview and under development. **
 
-Example - Push blob "hi.txt":
+Example - Push blob 'hi.txt' into a registry:
```
Comment (Wwwsylvia) [inline cmd/oras/blob/push.go:50]: doc: "into" or "to"? 🤔

File: cmd/oras/manifest/fetch.go [hunk 2]
```diff
@@ -87,17 +93,18 @@ Example - Fetch manifest with prettified json result:
...
-	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	target, err := opts.NewReadonlyTarget(ctx, opts.Common)
 	if err != nil {
 		return err
 	}
-
-	if repo.Reference.Reference == "" {
-		return oerrors.NewErrInvalidReference(repo.Reference)
+	if opts.Reference == "" {
+		return oerrors.NewErrInvalidReferenceStr(opts.Reference)
```
Comment (Wwwsylvia) [inline cmd/oras/manifest/fetch.go:101]: Just a question, what do the `opts.Reference` and `opts.RawReference` look like?

Comment (qweeah) [reply cmd/oras/manifest/fetch.go:101]: `opts.RawReference` is the full user input, e.g. `localhost:5000/test@sha256:xxxx`. `opts.Reference` is the tag or digest part, e.g. `sha256:XXXX`. Here should use RawReference.

Comment (qweeah) [reply cmd/oras/manifest/fetch.go]: Maybe I should move this error-checking function into target so it's less error-prone.

File: cmd/oras/blob/fetch.go [hunk 3]
```diff
@@ -48,17 +49,23 @@ func fetchCmd() *cobra.Command {
...
   oras blob fetch --output - localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5
 
 Example - Fetch and print the descriptor of a blob:
   oras blob fetch --descriptor localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5
 
-Example - Fetch the blob, save it to a local file and print the descriptor:
+Example - Fetch a blob, save it to a local file and print the descriptor:
   oras blob fetch --output blob.tar.gz --descriptor localhost:5000/hello@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5
+
+Example - Fetch and print a blob from OCI image layout folder 'layout-dir':
```
Comment (FeynmanZhou) [inline cmd/oras/blob/fetch.go:64]: Is it exact to use `layout-dir@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5` to represent an OCI layout folder?

File: cmd/oras/manifest/push.go [hunk 4]
```diff
@@ -75,19 +74,22 @@ Example - Push a manifest to repository 'locahost:5000/hello' and tag with 'tag1
 
 Example - Push a manifest to repository 'locahost:5000/hello' and tag with 'tag1', 'tag2', 'tag3' and concurrency level tuned:
   oras manifest push --concurrency 6 localhost:5000/hello:tag1,tag2,tag3 manifest.json
+
+Example - Push a manifest to an OCI layout folder 'layout-dir' and with tag 'latest':
```
Comment (FeynmanZhou) [inline cmd/oras/manifest/push.go:78]: Should we use `tag with 'latest'` to keep consistent with other examples?

**Review by FeynmanZhou — COMMENTED** [review]
The examples look good to me. Only two simple questions left.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/oras/blob/fetch.go (hunk 3 above)
Comment (qweeah) [reply cmd/oras/blob/fetch.go:64]: `layout-dir@sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5` means fetch a blob with digest `sha256:9a201d228ebd966211f7d1131be19f152be428bd373a92071c71d8deaf83b3e5` from an OCI image layout folder 'layout-dir'

File: cmd/oras/manifest/push.go (hunk 4 above)
Comment (qweeah) [reply cmd/oras/manifest/push.go:78]: I will update all tags to 'latest' to make it consistent.

Comment (qweeah) [reply cmd/oras/manifest/push.go:78]: Sorry I guess I misunderstand your meaning, will rephrase to `tag with`

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**